src/Interface/Modules/Render/ViewScene.h

```diff
@@ -72,15 +72,19 @@ namespace SCIRun {
       void cameraRotationChangeForwarder();
       void cameraLookAtChangeForwarder();
       void cameraDistnaceChangeForwarder();
+      void lockMutexForwarder();
       void mousePressSignalForTestingGeometryObjectFeedback(int x, int y, const std::string& selName);
 
 
     protected Q_SLOTS:
       //---------------- New Geometry --------------------------------------------------------------
       void updateModifiedGeometries();
+      void updateModifiedGeometriesAndSendScreenShot();
       void updateAllGeometries();
-      void newGeometryValue(bool forceAllObjectsToUpdate);
+      void newGeometryValue(bool forceAllObjectsToUpdate, bool pushScreenShotToState);
```

Comment:
The double-bool parameter list starts getting ugly. (client: "is it f(true, false) or f(false, true)???"). A labeled enum can be helpful, with the 4 options given distinct names.

Reply:
I guess you already have named methods that wrap the various bool-combinations. That will work as long as the `newGeometryValue` is private, which it probably can be here.